BinaryObjectScanner/FileType/ISO9660.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using BinaryObjectScanner.Data;
+using SabreTools.Data.Models.ISO9660;
+using SabreTools.IO.Extensions;
+
+namespace BinaryObjectScanner.FileType
+{
+    /// <summary>
+    /// ISO9660
+    /// </summary>
+    public class ISO9660 : DiskImage<SabreTools.Serialization.Wrappers.ISO9660>
+    {
+        /// <inheritdoc/>
+        public ISO9660(SabreTools.Serialization.Wrappers.ISO9660 wrapper) : base(wrapper) { }
+
+        /// <inheritdoc/>
+        public override string? Detect(Stream stream, string file, bool includeDebug)
+        {
+            // Create the output dictionary
+            var protections = new ProtectionDictionary();
+
+            // Standard checks
+            var subProtections
+                = RunISOChecks(file, StaticChecks.ISO9660CheckClasses, includeDebug);
+            protections.Append(file, subProtections.Values);
+
+            // If there are no protections
+            if (protections.Count == 0)
+                return null;
+
+            // Create the internal list
+            var protectionList = new List<string>();
+            foreach (string key in protections.Keys)
+            {
+                protectionList.AddRange(protections[key]);
+            }
+
+            return string.Join(";", [.. protectionList]);
+        }
+
+        // Checks whether the sequence of bytes is pure data (as in, not empty, not text, just high-entropy data)
+        public static bool IsPureData(byte[] bytes)
+        {
+            // Check if there are three 0x00s in a row. Two seems like pushing it
+            byte[] containedZeroes = {0x00, 0x00, 0x00};
+            int index = 0;
+            for (int i = 0; i < bytes.Length; ++i) {
```

Comment:
Formatting nitpick: Prefer opening curly brace on its own line if not using a single-line statement